### PR TITLE
mariadb 10.3: add a patch for debugging

### PR DESCRIPTION
**Pull request by komainu8**

### CODE REVIEW

File: packages/source/patches/mariadb-10.3.15-windows-build.diff [hunk 1]
```diff
@@ -0,0 +1,174 @@
+--- mariadb-10.3.15.orig/storage/innobase/srv/srv0start.cc	2019-05-14 03:13:10.000000000 +0900
++++ mariadb-10.3.15/storage/innobase/srv/srv0start.cc	2019-12-10 18:05:25.666910925 +0900
+@@ -2564,6 +2564,9 @@ void srv_shutdown_bg_undo_sources()
+ /** Shut down InnoDB. */
+ void innodb_shutdown()
+ {
++	ib::info() << "Start innodb_shutdown()."
++                   << "Shutdown operation " << srv_operation;
```
Comment (kou) [inline packages/source/patches/mariadb-10.3.15-windows-build.diff]:
```cpp
ib::info() << "Shutdown started; operation " << srv_operation;
```

File: packages/source/patches/mariadb-10.3.15-windows-build.diff [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
++
+ 	ut_ad(!my_atomic_loadptr_explicit(reinterpret_cast<void**>
+ 					  (&srv_running),
+ 					  MY_MEMORY_ORDER_RELAXED));
+@@ -2574,11 +2577,15 @@ void innodb_shutdown()
+ 	case SRV_OPERATION_RESTORE:
+ 	case SRV_OPERATION_RESTORE_DELTA:
+ 	case SRV_OPERATION_RESTORE_EXPORT:
++		ib::info() << "Start fil_close_all_files().";
```
Comment (kou) [inline packages/source/patches/mariadb-10.3.15-windows-build.diff]:
```cpp
ib::info() << "fil_close_all_files() started";
```

File: packages/source/patches/mariadb-10.3.15-windows-build.diff [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+ 		fil_close_all_files();
++		ib::info() << "End fil_close_all_files().";
```
Comment (kou) [inline packages/source/patches/mariadb-10.3.15-windows-build.diff]:
```cpp
ib::info() << "fil_close_all_files() completed";
```

File: packages/source/patches/mariadb-10.3.15-windows-build.diff [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+ 		break;
+ 	case SRV_OPERATION_NORMAL:
+ 		/* Shut down the persistent files. */
++		ib::info() << "Start shutdown the persistent files.";
```
Comment (kou) [inline packages/source/patches/mariadb-10.3.15-windows-build.diff]:
```cpp
ib::info() << "logs_empty_and_mark_files_at_shutdown() started";
```

File: packages/source/patches/mariadb-10.3.15-windows-build.diff [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+ 		logs_empty_and_mark_files_at_shutdown();
++		ib::info() << "End shutdown the persistent files.";
```
Comment (kou) [inline packages/source/patches/mariadb-10.3.15-windows-build.diff]:
```cpp
ib::info() << "logs_empty_and_mark_files_at_shutdown() completed";
```

File: packages/source/patches/mariadb-10.3.15-windows-build.diff [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+ 
+ 		if (ulint n_threads = srv_conc_get_active_threads()) {
+ 			ib::warn() << "Query counter shows "
+@@ -2588,20 +2595,26 @@ void innodb_shutdown()
+ 	}
+ 
+ 	/* Exit any remaining threads. */
++	ib::info() << "Start exit any remaining threads.";
```
Comment (kou) [inline packages/source/patches/mariadb-10.3.15-windows-build.diff]:
```cpp
ib::info() << "srv_shutdown_all_bg_threads() started";
```

File: packages/source/patches/mariadb-10.3.15-windows-build.diff [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+ 	srv_shutdown_all_bg_threads();
++	ib::info() << "End exit any remaining threads.";
```
Comment (kou) [inline packages/source/patches/mariadb-10.3.15-windows-build.diff]:
```cpp
ib::info() << "srv_shutdown_all_bg_threads() completed";
```

**Comment by komainu8** [comment]
@kou Thank you for your comments.
I've fixed them.

**Pull request merged**